assets/prototype/domain/eventEditor/DatesList.js

```diff
@@ -0,0 +1,18 @@
+import { useContext } from '@wordpress/element';
+import { DatesListDataContext } from './containers/datesListData/index';
+
+const DatesList = () => {
+	const list = useContext( DatesListDataContext );
```

Comment:
Here the data hook should be called nstead of context.

Reply:
I agree that for the dates list, we can just use a custom hook like `useDatesListData` which can simply query and return the data from the Apollo Client.

We will however use the Context API for the individual entities within a list, 
ie: each entity within the list will access a provider that reads and writes to the Apollo Client cache. The context for that _could_ still be wrapped in a hook if so desired, but I don't think that's required, it could work like what you've done here